impl/sql/mutationstorage/queue.go

```diff
@@ -46,27 +47,32 @@ func (m *Mutations) AddLogs(ctx context.Context, directoryID string, logIDs ...i
 }
 
 // Send writes mutations to the leading edge (by sequence number) of the mutations table.
+// Returns the logID/watermark pair that was written, or nil if nothing was written.
 // TODO(gbelvin): Make updates a slice.
-func (m *Mutations) Send(ctx context.Context, directoryID string, updates ...*pb.EntryUpdate) error {
+func (m *Mutations) Send(ctx context.Context, directoryID string, updates ...*pb.EntryUpdate) (*keyserver.WriteWatermark, error) {
 	glog.Infof("mutationstorage: Send(%v, <mutation>)", directoryID)
 	if len(updates) == 0 {
-		return nil
+		return nil, nil
 	}
 	logID, err := m.randLog(ctx, directoryID)
 	if err != nil {
-		return err
+		return nil, err
 	}
 	updateData := make([][]byte, 0, len(updates))
 	for _, u := range updates {
 		data, err := proto.Marshal(u)
 		if err != nil {
-			return err
+			return nil, err
 		}
 		updateData = append(updateData, data)
 	}
 	// TODO(gbelvin): Implement retry with backoff for retryable errors if
 	// we get timestamp contention.
-	return m.send(ctx, time.Now(), directoryID, logID, updateData...)
+	ts := time.Now()
```

Comment:
Looks dangerous to me, as servers have time skew. Shouldn't the returned watermark be based on what the storage thinks the "time" is? Consider returning a "strong" watermark from the `send` call below.

Reply:
The write implementation first verifies that the timestamp provided is greater than any other timestamp currently in the database.  A server that has a slow clock will simply fail more often when racing another writer that has a clock further ahead.

Reply:
I provide the watermark here instead of returning it so that I can have meaningful unit tests.

Reply:
Why does returning the same watermark that [this](https://github.com/google/keytransparency/blob/edd2c90f078bec0d40021e5c076e36c9899fb2e1/impl/sql/mutationstorage/queue.go#L138) line generates in `m.send` not work?

Reply:
(or just the timestamp)

Reply:
That line doesn't query the current time, it just convertes the passed in `ts time.Time` into an integer `ts.UnixNano()`

Reply:
Ah, indeed.